android/app/src/main/java/com/example/huntertalk/LeaveGroupPopUp.java

```diff
@@ -70,6 +80,11 @@ public void onCancelled(@NonNull DatabaseError databaseError) {
                     }
                 });
 
+                //set finish to finish if user choose to leave group so that inside group activity finishes
+                finish = "finish";
+
+                intent = new Intent(LeaveGroupPopUp.this, InsideGroupActivity.class);
```

Comment:
This doesn't actually close InsideGroupActivity, it only starts a new Home_Page, and the old insideGroupActivity is still running in the background.